components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -5022,4 +5023,41 @@ public Map<String, String> getIdPNamesById(int tenantId, Set<String> idpIds)
             IdentityDatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
         }
     }
+
+    /**
+     * Get IDP group data by IDP group IDs.
+     *
+     * @param idpGroupIds List of IDP group IDs.
+     * @param tenantId    Tenant ID.
+     * @return List of IDP groups.
+     * @throws IdentityProviderManagementException If an error occurred while retrieving IDP groups.
+     */
+    public List<IdPGroup> getIdPGroupsByIds(List<String> idpGroupIds, int tenantId)
+            throws IdentityProviderManagementException {
+
+        String query = IdPManagementConstants.SQLQueries.GET_IDP_GROUPS_BY_IDP_GROUP_IDS;
+        String placeholders = String.join(",", Collections.nCopies(idpGroupIds.size(), "?"));
```

Comment:
lets return an empty list if idpGroupIds is empty. This will avoid possible syntax issues with empty list.
Eventhough the service layer has handed that case, there could be possibility to occur the issues if db method invoked from another db method in the class

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5143/commits/5b7138f04fb974349e2e0a3157b9ce2109f58313